main.py

```diff
@@ -0,0 +1,1605 @@
+    output_messages(message_array, curr_msg_topic)
+
+def insert_treeview_topic():
+    global curr_msg_topic
+
+    tab3_topic = [] #list for topics in Treeview in Tab3
+    
+    if topic not in tab3_topic:
+        if topic == None:  #for initial subscriptions in the beginning of program
+            short_topic = '...' + address[-10:]  #only show the last 10 characters of a topic
+            curr_msg_topic = address
+            tab3_tree.insert("", END, iid=curr_msg_topic, text=short_topic)
+            tab3_topic.append(address)
+            
+            print('1:', tab3_topic)
+            print('address:', address)
+        else:
+            short_topic = topics.shorten_topic(topic)
+            short_topic = '...' + topic[-10:]
+            tab3_tree.insert("", END, iid=curr_msg_topic, text=short_topic)     #parent
+            tab3_topic.append(topic)
+            
+            print('2:', tab3_topic)
+            print('topic:', topic)
+
+def output_messages(message_array, curr_msg_topic):
+    global j 
+
+    #insert messages in Treeview in its appropriate topic section
+    curr_msg_topic = str(curr_msg_topic)
+    for i in message_array:
+        tab3_tree.insert(parent=curr_msg_topic, index=j, values=i)    #child, insert from top
+        j = j + 1
+
+    '''
+    i = tab3_tree.insert("", END, text="Topic1")
+    #this could be the data topic
+    tab3_tree.insert(parent=i, index=0, iid=0, values=('SubTopic1', 5, 'Time1'), tags='odd')
+    #this could be a data value
+    tab3_tree.insert(parent=i, index=1, iid=1, values=('SubTopic2', 10, 'Time2'), tags='even')
+    j = tab3_tree.insert("", END, text="Topic2")
+    tab3_tree.insert(parent=j, index=2, iid=2, values=('SubTopic3', 3, 'Time3'), tags='odd')
+    '''
+
+'''Tab 2 Stuff'''
+def tab2_update_msgBox(data):   #add list of messages into box
+    tab2_messages_list.delete(0,END)
+    for item in data:
+        tab2_messages_list.insert(END, item)
+
+def tab2_update_listBox(data):   #update the listbox
+    tab2_listBox.delete(0,END)    #clear the listbox
+    for item in data:   #add topics to listbox
+        tab2_listBox.insert(END, item)
+
+def do_clear2():
+    tab2_topic_input.delete(0,END)  #clear entry box
+    tab2_msg_input.delete(0,END) #clear entry box
+    event_clicks.tab2_insert_shadow_text_left(tab2_topic_input)   #put shadow texts back in
+    event_clicks.tab2_insert_shadow_text_right(tab2_msg_input)
+    
+def tab2_fillout_msg(e):
+    tab2_msg_input.delete(0,END)
+    tab2_msg_input.insert(0, tab2_messages_list.get(ANCHOR))
+    tab2_msg_input.config(fg='black')
+
+def tab2_fillOut(e):
+    tab2_topic_input.delete(0,END)    #delete whatever is in entry box
+    tab2_topic_input.insert(0, tab2_listBox.get(ANCHOR))  #add clicked list item to listbox
+    tab2_topic_input.config(fg='black')
+
+'''Tab 1 Stuff'''
+def do_clear():
+    tab1_sub_to_topic.delete(0,END)
+    event_clicks.tab1_insert_shadow_text(tab1_sub_to_topic) #insert shadow text
```

Comment:
One downside of the closure route I suggest in another comment is that you would lose direct access to the clear/show functions. Were you to go this route, you could use `tab1_sub_to_topic.event_generate("<<FocusOut>>", when="tail")` as a work-around, but since you do need the ability to clear these types of inputs due to external stimulus, at that point I think that justifies just straight-up subclassing Input class.